web/src/hooks/useLogActionsExtension.tsx

```diff
@@ -11,21 +12,73 @@ const useLogActionsExtension: ExtensionHook<Array<Action>, LogActionsExtensionOp
   options,
 ) => {
   const { t } = useTranslation('plugin__logging-view-plugin');
+  const [actions, setActions] = React.useState<Action[]>([]);
 
-  // TODO: transform promQL into logQL
-  const alertQuery = options.alert?.rule?.query ?? '';
-  const href = `/monitoring/logs?q=${alertQuery}`;
-  const [actions] = React.useState([
-    {
-      id: 'link-to-logs',
-      label: (
-        <>
-          <ListIcon /> {t('See related logs')}
-        </>
-      ),
-      cta: { href },
-    },
-  ]);
+  const alertingRuleName = options.alert?.rule.name;
+
+  useEffect(() => {
+    if (alertingRuleName) {
+      const { request, abort } = listGoals({
+        goalsRequest: {
+          start: {
+            class: 'alert.alert',
+            queries: [{ Labels: { alertname: alertingRuleName } }],
+          },
+          goals: ['application.log', 'infrastructure.log', 'audit.log'],
+        },
+      });
+
+      request()
+        .then((response) => {
+          let logQL: string;
+          let tenant: string;
+
+          response.forEach((goal) => {
+            goal?.queries?.forEach(({ count, query }) => {
+              if (query.LogQL && count > 0) {
+                logQL = query.LogQL;
+                if (goal.class === 'application.log') {
+                  tenant = 'application';
+                } else if (goal.class === 'audit.log') {
+                  tenant = 'audit';
+                } else if (goal.class === 'infrastructure.log') {
+                  tenant = 'infrastructure';
+                }
+              }
+
+              if (logQL && tenant) {
```

Comment:
This probably can be outside the `response.forEach`, I assume we are searching in the response for the first goal that matches logs.

Reply:
Thanks @jgbernalp. I tidied up the function and made it more clear that we're just using the first goal that matches logs with this PR.